Neos.Cache/Classes/Backend/SimpleFileBackend.php

```diff
@@ -467,4 +460,74 @@ protected function configureCacheDirectory()
 
         $this->cacheDirectory = $cacheDirectory;
     }
+
+    /**
+     * Reads the cache data from the given cache file, using locking.
+     *
+     * @param string $cacheEntryPathAndFilename
+     * @param int|null $offset
+     * @param int|null $maxlen
+     * @return boolean|string The contents of the cache file or FALSE on error
+     */
+    protected function readCacheFile($cacheEntryPathAndFilename, $offset = null, $maxlen = null)
+    {
+        for ($i = 0; $i < 3; $i++) {
+            $data = false;
+            try {
+                $file = fopen($cacheEntryPathAndFilename, 'rb');
+                if ($file === false) {
+                    continue;
+                }
+                if (flock($file, LOCK_SH) !== false) {
+                    if ($offset !== null) {
+                        fseek($file, $offset);
+                    }
+                    $data = fread($file, $maxlen !== null ? $maxlen : filesize($cacheEntryPathAndFilename) - (int)$offset);
+                    flock($file, LOCK_UN);
+                }
+                fclose($file);
+            } catch (\Exception $e) {}
+
+            if ($data !== false) {
+                return $data;
+            }
+            usleep(rand(10, 500));
+        }
+
+        return $data;
+    }
+
+    /**
+     * Writes the cache data into the given cache file, using locking.
+     *
+     * @param string $cacheEntryPathAndFilename
+     * @param string $data
+     * @return boolean|integer Return value of file_put_contents
+     */
+    protected function writeCacheFile($cacheEntryPathAndFilename, $data)
+    {
+        for ($i = 0; $i < 3; $i++) {
+            // This can be replaced by a simple file_put_contents($cacheEntryPathAndFilename, $data, LOCK_EX) once vfs
+            // is fixed for file_put_contents with LOCK_EX, see https://github.com/mikey179/vfsStream/wiki/Known-Issues
+            $result = false;
+            try {
+                $file = fopen($cacheEntryPathAndFilename, 'wb');
+                if ($file === false) {
```

Comment:
can we add a comment which case this is? I am currently not sure in which (race-condition) case this might happen.